Added a New Provider Component - Google Secrets Provider (#39790)
open-telemetry/opentelemetry-collector-contrib: confmap/provider/googlesecretsprovider/README.md

> # Google Secrets Provider

Reviewer: Throughout this document, in place of `Otel` and `Otel Collector` let's just say `collector`. Additionally, the audience for this document is the user themselves, so we can use second party language like "you/your".

| Stability     | [development]  |
| Distributions | [] |
| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Aprovider%2Fgooglesecretsprovider%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Aprovider%2Fgooglesecretsprovider) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Aprovider%2Fgooglesecretsprovider%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Aprovider%2Fgooglesecretsprovider) |
| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@XuechunHou](https://www.github.com/XuechunHou) |

[development]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-stability.md#development
<!-- end autogenerated section -->

## Summary

> This Provider component offers Otel users a secure way to reference secrets or sensitive information in their Otel Collector configurations using [Google Secret Manager](https://cloud.google.com/security/products/secret-manager). Users place placeholders in the format `${googlesecretsprovider:projects/<project Id>/secrets/<secret Id>/versions/<version Id>}` within their configurations. The actual secrets will then be fetched dynamically from [Google Secret Manager](https://cloud.google.com/security/products/secret-manager) during Otel Collector initialization.

Reviewer: ```suggestion
This Provider component offers a secure way to reference secrets or sensitive information in collector configurations using [Google Secret Manager](https://cloud.google.com/security/products/secret-manager). Use a placeholder in the format `${googlesecretsprovider:projects/<project Id>/secrets/<secret Id>/versions/<version Id>}` within your configuration. The actual secrets will then be fetched dynamically from [Google Secret Manager](https://cloud.google.com/security/products/secret-manager) during collector initialization.
```

> ## How it works

Reviewer: This is a usage guide, but it doesn't explain how it works.
```suggestion
## Usage
```

- Simply replace plaintext secrets within the Otel configurations with the placeholder: `${googlesecretsprovider:projects/<project Id>/secrets/<secret Id>/versions/<version Id>}`

An example Otel configuration:

```
receivers:
  otlp:
    protocols:
      grpc:
      http:
processors:
  batch:

exporters:
  logging:
    loglevel: debug
  http:
    endpoint: "https://example.com/api/metrics"
    headers:
      X-API-Key: ${googlesecretsprovider:projects/12345/secrets/my-secret/versions/1}
service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, http]
    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, http]
    logs:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, http]

```

> ### Prerequisites

Reviewer: This section makes no mention of authentication, surely there needs to be some local authentication via [Application Default Credentials](https://cloud.google.com/docs/authentication/provide-credentials-adc)?

Author: Added.
